public/install/functions.php

```diff
@@ -17,126 +25,196 @@ function checkPhpVersion()
-    $str = substr($str, 0, -1);
-
-    $fp = fopen($envFile, 'w');
-    fwrite($fp, $str);
-    fclose($fp);
+    $str = "{$envKey}={$envValue}";
+    return putenv($str);
 }
 
-function getEnvironmentValue($envKey)
+/**
+ * Gets the variable from the env file
+ * @param string $envKey The environment variable to look for
+ * @return array|false|string Returns the value if found, otherwise returns false.
+ */
+function getEnvironmentValue(string $envKey): array|false|string
 {
-    $envFile = dirname(__FILE__, 3).'/.env';
-    $str = file_get_contents($envFile);
+    return getenv($envKey);
+}
+
 
-    $str .= "\n"; // In case the searched variable is in the last line without \n
-    $keyPosition = strpos($str, "{$envKey}=");
-    $endOfLinePosition = strpos($str, PHP_EOL, $keyPosition);
-    $oldLine = substr($str, $keyPosition, $endOfLinePosition - $keyPosition);
-    $value = substr($oldLine, strpos($oldLine, '=') + 1);
+/**
+ * Encrypt the given value
+ * @param mixed $value The variable to be encrypted
+ * @param bool $serialize If the encryption should be serialized
+ * @return string Returns the encrypted variable.
+ */
+function encryptSettingsValue(mixed $value, $serialize = true): string
+{
+    $appKey = getEnvironmentValue('APP_KEY');
+    $appKey = base64_decode(Str::after($appKey, 'base64:'));
+    $encrypter = new Encrypter($appKey, 'AES-256-CBC');
+    $encryptedKey = $encrypter->encrypt($value, $serialize);
+
+    return $encryptedKey;
+}
+
+/**
+ * Decrypt the given value
+ * @param mixed $payload The payload to be decrypted
+ * @param bool $unserialize If the encryption should be unserialized
+ * @return mixed Returns the decrypted variable on success, throws otherwise.
+ */
+
+function decryptSettingsValue(mixed $payload, $unserialize = true)
+{
+    $appKey = getEnvironmentValue('APP_KEY');
+    $appKey = base64_decode(Str::after($appKey, 'base64:'));
+    $encrypter = new Encrypter($appKey, 'AES-256-CBC');
+    $decryptedKey = $encrypter->decrypt($payload, $unserialize);
 
-    return $value;
+    return $decryptedKey;
 }
 
-function run_console($command)
+/**
+ * Run a shell command
+ * @param string $command The command string to run
+ * @return false|string|null Returns the result from the command.
+ */
+function run_console(string $command)
 {
     $path = dirname(__FILE__, 3);
     $cmd = "cd '$path' && bash -c 'exec -a ServerCPP $command' 2>&1";
 
     return shell_exec($cmd);
 }
 
-function wh_log($log_msg)
+/**
+ * Log to installer.log in the install folder
+ * @param string $log_msg the message to log
+ * @return void No output.
+ */
+function wh_log(string $log_msg)
 {
     $log_filename = 'logs';
-    if (! file_exists($log_filename)) {
+    if (!file_exists($log_filename)) {
         // create directory/folder uploads.
         mkdir($log_filename, 0777, true);
     }
-    $log_file_data = $log_filename.'/installer.log';
+    $log_file_data = $log_filename . '/installer.log';
```

Comment:
When i think about, we should put the installer log at the same place where the laravel log lives in. (storage/logs)

Reply:
I agree.